draft-ietf-ohai-ohttp.md

```diff
@@ -444,8 +447,8 @@ The Nenc parameter corresponding to the KEM used in HPKE can be found in
 {{Section 7.1 of !HPKE}}.  Nenc refers to the size of the encapsulated KEM
 shared secret, in bytes.
 
-An encrypted HTTP response includes a binary-encoded HTTP message {{BINARY}}
-and no other content; see {{fig-res-pt}}.
+By default, an encrypted HTTP response includes a binary-encoded HTTP
```

Comment:
I'd prefer the original here - it's more direct.

The use of an alternative content type requires a bit more than just pointing to some advice about how to switch out the content.  That is, the repurposing section is information about how someone might start to build a new, but similar encapsulation format.

Reply:
I think we ought to be clear that the inner content type can change as applications need it to change, whether that's done here or some other way. Repurposing the format should only require a label change and nothing else. If there's more that needs to be done, then we should update the repurposing section accordingly.

Reply:
I agree with Chris. Since the "new encapsulation format" is just a matter of swapping the media type in the info strings and change the BHTTP content to another type, it's useful to be clear in the OHTTP spec which parts are the parts to swap out, and which are the parts that are constant across any contained media type.

I don't think more needs to be done — if there is something else, I'd like to hear it.

Reply:
This particular change is fine.  But the reuse thing is amply covered by the section; I just don't think we need quite so many references to it.

FWIW, repurposing requires defining a new media type (`message/whatever-except-encrypted`) and then describing how that uses this document with a label based on the encapsulated type (`message/whatever`) instead of `message/bhttp`.

Reply:
We can cut down the references if there's still some text to make it clear that this isn't _always_ Binary HTTP.

Also, regarding the requirements for using things other than BHTTP: currently, the text just says that you need to have the inner media type be used in the label for the info. I don't think there's any requirement that other payloads not use message/ohttp-req and message/ohttp-res for the outer types. They certainly _can_ define other media types for the outside if they want, particularly if two different types are supported by the same gateway. However, if the client knows about the gateway via out-of-band configuration, it could simply know the inner format to use and not need to have a relay in between know anything about the inner encapsulated payload being something other than binary HTTP.

Reply:
Hmm.  This is partly why I made #91.  The content of this media type does not identify what it carries - including whether it contains a request or a response, which is why we define two - so you really can't use message/ohttp-re[qs] for anything other than a binary HTTP message.